### PR TITLE
Add regex-tdfa to preloaded packages

### PR DESCRIPTION
**Pull request by asarkar**
regex-tdfa is a fairly common package, and other tracks do allow for use of regex, so it's not philosophically banned either. It is very useful for several exercises.

### CODE REVIEW

**Comment by asarkar** [comment]
Related to https://github.com/exercism/haskell/issues/1045

**Comment by petertseng** [comment]
yes, I confirm there is no ban on regex.

**Comment by petertseng** [comment]
yes, I confirm there is no ban on regex. To be clear: I explicitly express my approval of this PR. The only reason I'm not Approving in the GitHub sense of the word is that I consider myself to have no power here (even though I have write permissions in this repository, the previous sentence is meant to express the social factor governed by self-regulation, rather than the technical factor governed by permissions/technology)

**Review by petertseng — APPROVED** [review]
I changed my mind and there are a variety of reasons that I should Approve in the GitHub sense of the word (the reasons boil down to visibility and ease of understanding my viewpoint at a glance when reviewing history). My previous statements do of course still apply!

**Comment by petertseng** [comment]
Ah something I'm thinking about... it absolutely makes sense that for changes to the available packages that it'd want a track maintainer to weigh in on whether adding that package is reasonable. Given that's true, would it make sense to have some configuration so that track maintainers are automatically added to PRs that change `pre-compiled/package.yaml`?

**Comment by petertseng** [comment]
Moving above comment to its own separate issue

**Comment by petertseng** [comment]
From my point of view, we have everything we need from you at this moment, but rest assured that we stand ready to let you know if that happens to change.

However, I would be far more hesitant to be discourteous to the organisations admins by calling them bots.

**Comment by petertseng** [comment]
I owe an explanation here - the reason I have not merged this is I believe my merge has no power. Even if I merged it, I cannot update the test runner, so it's better for me to leave the merging to someone who has the ability to make that happen.

If there needs to be a process improvement so that a track maintainer can do this, we could have a discussion on what changes would need to be made so that this is possible, but I've usually been taught to go with the principle of least privilege, and giving me the power to break something that I really should not be allowed to break is an uncomfortable prospect.

**Review by ErikSchierboom — APPROVED** [review]
I'm happy to be the one making the decision on whether to merge this or not.

This looks good to me! Thanks @asarkar

**Comment by asarkar** [comment]
Now that this PR is merged, can I start using this package, or do I have to wait for a release?

**Comment by ErikSchierboom** [comment]
@asarkar The only thing you had to wait for was the deploy workflow, which usually finishes in 10-20 minutes after merging a PR: https://github.com/exercism/haskell-test-runner/actions/runs/2678146178 So, yes: you can use it.

**Pull request merged**
